### PR TITLE
Kernel: Add new integer variable type to SysFS

### PR DESCRIPTION
**Pull request by Optimoos**
This change adds an integer variable type to SysFS in preparation for allowing certain features to be configured live, such as network TTL.

### CODE REVIEW

File: Kernel/FileSystem/SysFS/Subsystems/Kernel/Variables/IntegerVariable.cpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+ErrorOr<void> SysFSSystemIntegerVariable::try_generate(KBufferBuilder& builder)
+{
+    return builder.appendff("{}\n", value());
+}
+
+ErrorOr<size_t> SysFSSystemIntegerVariable::write_bytes(off_t, size_t count, UserOrKernelBuffer const& buffer, OpenFileDescription*)
+{
+    MutexLocker locker(m_refresh_lock);
+    // Note: We do all of this code before taking the spinlock because then we disable
```
Comment (supercomputer7) [inline Kernel/FileSystem/SysFS/Subsystems/Kernel/Variables/IntegerVariable.cpp:21]: Is this really the case where page fault will not work? I know `SysFSSystemStringVariable` mentioned this, but maybe we could just use `safe_memcpy` here (and in `SysFSSystemStringVariable` too).

File: Kernel/FileSystem/SysFS/Subsystems/Kernel/Variables/IntegerVariable.cpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        }
+        value++;
+    }
+
+    result *= sign;
+
+    // NOTE: If we are in a jail, don't let the current process to change the variable.
+    if (Process::current().is_currently_in_jail())
+        return Error::from_errno(EPERM);
+    dbgln("Setting value: {}", result);
```
Comment (supercomputer7) [inline Kernel/FileSystem/SysFS/Subsystems/Kernel/Variables/IntegerVariable.cpp:49]: Please either remove this, or put it behind a debug flag.

File: Kernel/FileSystem/SysFS/Subsystems/Kernel/Variables/IntegerVariable.cpp (hunk 1 above)
Comment (supercomputer7) [reply Kernel/FileSystem/SysFS/Subsystems/Kernel/Variables/IntegerVariable.cpp:21]: Also, I think this comment is old and not relevant even for `SysFSSystemStringVariable` anymore too.

**Comment by Optimoos** [comment]
It looks like that note came from the original boolean variable implementation, and has been copied along since. I can remove it from this for sure.

Re `safe_memcpy`, as I'm not 100% clear on where locks are required, just to be sure - you're saying that the mutex probably isn't required and that doing safe_memcpy on the result should be good enough?

I've done the other cleanup as well, but will wait on the `safe_memcpy` discussion before I push the update.

**Comment by supercomputer7** [comment]
> It looks like that note came from the original boolean variable implementation, and has been copied along since. I can remove it from this for sure.
> 
> Re `safe_memcpy`, as I'm not 100% clear on where locks are required, just to be sure - you're saying that the mutex probably isn't required and that doing safe_memcpy on the result should be good enough?
> 
> I've done the other cleanup as well, but will wait on the `safe_memcpy` discussion before I push the update.

If we do `safe_memcpy`, then it doesn't matter if we took a spinlock or not - page faults could still occur, hence everything is fine. The initial problem (I think) was that I tried to use mutexes to protect `/sys/kernel/...` variables, but since we use spinlocks, that problem is not relevant.

**Pull request closed without merging**